api/v1alpha1/tiflowcluster_types.go

```diff
@@ -512,59 +501,135 @@ type StorageVolumeStatus struct {
+	// transitioned from one to another
+	// +required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+}
+
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen=true
+
+// ExecutorStatus defines the desired state of tiflow executor
 type ExecutorStatus struct {
-	Synced         bool                      `json:"synced"`
-	Phase          MemberPhase               `json:"phase,omitempty"`
+	Image          string                    `json:"image,omitempty"`
 	StatefulSet    *apps.StatefulSetStatus   `json:"statefulSet,omitempty"`
 	Members        map[string]ExecutorMember `json:"members,omitempty"`
 	PeerMembers    map[string]ExecutorMember `json:"peerMembers,omitempty"`
 	FailureMembers map[string]ExecutorMember `json:"failureMembers,omitempty"`
 	FailoverUID    types.UID                 `json:"failoverUID,omitempty"`
-	Image          string                    `json:"image,omitempty"`
 	// Volumes contains the status of all volumes.
 	Volumes map[string]*StorageVolumeStatus `json:"volumes,omitempty"`
+	// LastUpdateTime means the time when the status of Executor cluster's info was updated
+	// +required
+	LastUpdateTime metav1.Time `json:"lastUpdateTime"`
+
+	// Phase represents the observed state of a tiflow executor
+	// Update by sync types
+	// +required
+	Phase ExecutorPhaseType `json:"phase,omitempty"`
+	// (Optional) Message related to the status of the ExecutorCluster
+	// +optional
+	Message string `json:"message,omitempty"`
+	// +nullable
+	SyncTypes []ClusterSyncType `json:"syncTypes,omitempty"`
+	// LastTransitionTime means the time when the status of Executor Phase
+	// transitioned from one to another.
+	// +required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
 }
 
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen=true
+
 // TiflowClusterCondition is tiflow cluster condition
 type TiflowClusterCondition struct {
 	// Type of the condition.
-	Type string `json:"type"`
+	Type TiflowClusterConditionType `json:"type"`
 	// Status of the condition, one of True, False, Unknown.
-	Status corev1.ConditionStatus `json:"status"`
-	// The last time this condition was updated.
-	// +nullable
-	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty"`
-	// Last time the condition transitioned from one status to another.
+	Status metav1.ConditionStatus `json:"status"`
+	// LastTransitionTime means last time the condition transitioned from one status to another.
 	// +nullable
 	// +optional
 	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
-	// The reason for the condition's last transition.
-	// +optional
-	Reason string `json:"reason,omitempty"`
-	// A human readable message indicating details about the transition.
-	// +optional
-	Message string `json:"message,omitempty"`
 }
 
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen=true
+
 // TiflowClusterStatus defines the observed state of TiflowCluster
 type TiflowClusterStatus struct {
+	// +required
+	ServerName string `json:"serverName"`
```

Comment:
Can we change it to `serverAddress` and put it into `MasterStatus`?

Reply:
done